### ISSUE TITLE
Code of Conduct

### ISSUE THREAD

**Issue by sonneveld** [adventuregamestudio/ags#885]
I would like to get people's thoughts on introducing some sort of Code of Conduct.

I'm not interested in silencing discussion but I would like people to feel good about contributing to this project, with everyone playing nice and being respectful of each other.

For reference, here are the rules for the AGS discord:

```
We have a few rules, taken from the AGS forums, that everyone is expected to abide by. Please note, you must be 13 years or older to use Discord. 
The main AGS site can be found at: https://www.adventuregamestudio.co.uk/
1) Be respectful. We have people from various backgrounds, belief systems and industries in here. Some of us are professionals, some are hobbyists, but we should all strive to get along under the AGS banner as much as possible. Additionally, hate speech of any kind will not be tolerated. 
2) Linking or sending harmful material such as viruses and malware will result in immediate and permanent bans. Running bots or scripts without permission from the moderator team can result in being banned - please discuss it with the team. 
3) Use proper grammar and spelling when possible. AGS Discord and Forums are primarily meant for English users and we recognize that many of our members do not speak English as a first (or second!) language. Please keep this in mind when using these channels.
4) Please do not use @ mentions frivolously. Moderators are always available to take private messages.
5) Post in the proper channels. We have channels for most of the basic topics here. If you are unsure feel free to ask in General. If you have suggestions for new channels let a moderator know.
Unsure of any of these rules or just want to ask a question? Don't hesitate to reach out to our moderator team. 
```

The AGS Forums: https://www.adventuregamestudio.co.uk/forums/index.php?topic=45786.0

and finally, the recently introduced Code of Conduct for the Linux kernel: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=8a104f8b5867c682d994ffa7a74093c54469c11f

**Comment by morganwillcock** [comment]
The base of the Linux kernel one is this one:
https://www.contributor-covenant.org/version/1/4/code-of-conduct.html

I guess exact interpretation will always have a subjective element to it, but I think having something like this is a good idea. If nothing else, I would like to reassure people that if they had a go at implementing something it would be a positive experience, regardless of whether changes were accepted or not.

**Comment by fernewelten** [comment]
It's nice to have some sort of a charter, but that by itself is probably not going to cut it. Declaring a standard and living by it are two entirely different things, otherwise we wouldn't have racial discrimination in the USA. They declared that they wouldn't have that more than two centuries ago.

From a practical point of view, how are we going to make happen what we want to declare? After all, a lot of  newcomer that consider joining us will probably ignore the declarations – they'll directly flip through some sample discussions and bug handlings to determine how we really do interact and how we resolve things in practice that have gone awry.

**Comment by morganwillcock** [comment]
It isn't meant to magically stop bad things happening on the Internet, but if there is a problem you can:
- demonstrate that there are guidelines and show that they are reasonably visible
- point someone at the guidelines when they aren't staying within them
- give a clear explaination why someone is blocked (temporarily or otherwise)

**Comment by fernewelten** [comment]
From a user's point of view, if we agree on a standard such as https://www.contributor-covenant.org/version/1/4/code-of-conduct.html , everyone can go “Ah, **that** standard, I know that one“ and go on with their coding life with minimum hassle. That might be an advantage over home-brew, where everyone would need to peruse the details to find out whether something unexpected is in it.

**Comment by morganwillcock** [comment]
The only sticking point of selecting an existing one (like this particular one) is that it assumes that there is some kind of 'admin' contact point that someone can go to in the event of a problem. So it would either need to be a little more general or someone needs to maintain a domain for e-mail addresses etc. I'm not 100% sure who will be around long term to look after things, so perhaps something broader (similar to the Discord guidelines) would be preferable?

**Comment by gurok** [comment]
I don't think it's warranted. Who has upset you with his or her conduct, @sonneveld ?

**Comment by sonneveld** [comment]
> I don't think it's warranted. Who has upset you with his or her conduct, @sonneveld ?

My intent wasn't to single anyone out here. I wanted to try to be transparent with any moderation to ensure kindness and respect. (I found this a good summary of what I mean: https://snarky.ca/setting-expectations-for-open-source-participation/ ) 

If people are happy with admin discretion, I will close this ticket and be open with any moderation decisions from the AGS github admins.

**Comment by fernewelten** [comment]
So, this has been open for  three months; last statement is two months old.

Are we going to do something specific about the matter, or do we consider it resolved for lack of interest?

**Comment by AlanDrake** [comment]
It's been more than a year, I'll close this issue.

**Comment by ericoporto** [comment]
addressed by #2606
